app/models/manager_refresh/inventory/persister.rb

```diff
@@ -12,6 +12,28 @@ def initialize(manager, target)
     initialize_inventory_collections
   end
 
+  def self.has_inventory(options)
+    # create method on class
+    name = options[:association]
+    define_method(name) do
+      collections[name] ||= begin
+        options[:parent] ||= manager
+        collection = ::ManagerRefresh::InventoryCollection.new(options)
+        if options[:builder_params]
+          options[:builder_params].transform_values! do |value|
+            if value.respond_to? :call
+              value.call(self)
+            else
+              value
+            end
+          end
+        end
+        yield(collection, self) if block_given?
+        collection
```

Comment:
I think this is right.

But every time I read this I keep wondering if we want to return the block results instead:

```
(block_given? ? yield(collection, self) : nil) || collection
```